### ISSUE TITLE
Consider adding a `sphinx_gallery_expected_error` code-block option to show an expected error

### ISSUE THREAD

**Issue by smarie** [sphinx-gallery/sphinx-gallery#912]
( From https://github.com/smarie/mkdocs-gallery/issues/23 )

Sometimes we write a code block to show that something raises an error.
It is a bit tideous to have to try/catch this error to prevent mkdocs-gallery to fail.
It would be much better to be able to say that we expect it:

```python
"""
Example
=======

The following code raises an error
"""
# sphinx_gallery_expected_error
1 + "hello"

# %%
# But this works:

1 * "hello"
```

We could maybe have a `sphinx_gallery_expected_error` similar to the `sphinx_gallery_defer_figures` configuration. 
It could even accept an optional argument (the exception type, as in `pytest.raises`) to avoid being too tolerant.

**Comment by larsoner** [comment]
Agreed this would be useful. @smarie do you want to try implementing this?

**Comment by smarie** [comment]
I will see if I can find an hour next week. If not that will have to wait one or two more weeks - holidays ahead ;)
